lib/jazz_fingers.rb

```diff
@@ -28,7 +28,24 @@ def prompt
           separator: config.prompt_separator,
           application_name: config.application_name
         )
-      @prompt.config
+    end
+
+    # Add JazzFingers prompt to the Pry::Prompt hash to enable changing it with
+    # `change-prompt`.
+    #
+    # Return the Pry::Prompt object.
+    def jazz_fingers_prompt
+      return Pry::Prompt[:jazz_fingers] if Pry::Prompt[:jazz_fingers]
+
+      Pry::Prompt.add(
+        :jazz_fingers,
+        "A spruced-up prompt provided by jazz_fingers.",
+        prompt.separators
+      ) do |context, _nesting, pry, separator|
+        prompt.template(Pry.view_clip(context), pry, separator)
+      end
+
+      Pry::Prompt[:jazz_fingers]
```

Comment:
What if we make this in the `Prompt` class instead?

There is a lot more going on on the `prompt` front then just `template(Pry.view_clip(context), pry, separator)`.

What if the `Prompt` class becomes responsible to add itself to `Pry::Prompt`?

Reply:
> There is a lot more going on on the prompt front then just template(Pry.view_clip(context), pry, separator)

Can you disambiguate? I'm not sure what you mean by this.

> What if we make this in the Prompt class instead?

That's possible (that would be similar to 594223d). The current approach takes the `JazzFingers` module as the integration point with `Pry`, so the hard dependency on the `Pry` class is isolated there.